### PR TITLE
CBG-1023: Error if custom conflict set without func

### PR DESCRIPTION
**Pull request by JRascagneres**
If custom conflict resolution type is set but no function is provided return an error. 
Also decided to add a conflict resolution type validity check.

Tested with a unit test attempting to put a conflict type of custom without func. 

---

Also had to fix a couple of tests which threw errors given the new validation

### CODE REVIEW

File: db/sg_replicate_cfg.go [hunk 1]
```diff
@@ -163,6 +163,14 @@ func (rc *ReplicationConfig) ValidateReplication(fromConfig bool) (err error) {
 		}
 	}
 
+	if !rc.ConflictResolutionType.IsValid() {
+		return base.HTTPErrorf(http.StatusBadRequest, "Conflict resolution type is invalid")
```
Comment (bbrks) [inline db/sg_replicate_cfg.go]: Include valid types in error message, similar to the `direction` validation. Should define in a constant for easier test assertion.

File: rest/replication_api_test.go [hunk 2]
```diff
@@ -1521,6 +1521,40 @@ func TestSGR1CheckpointMigrationPush(t *testing.T) {
 	assert.Equal(t, int64(1), r.Push.Checkpointer.Stats().GetCheckpointMissCount)
 }
 
+func TestReplicationConfig(t *testing.T) {
```
Comment (bbrks) [inline rest/replication_api_test.go]: Might be worth just adding extra test cases in `TestValidateReplication` instead of a new test?

**Review by bbrks — CHANGES_REQUESTED** [review]

File: rest/replication_api_test.go (hunk 2 above)
Comment (JRascagneres) [reply rest/replication_api_test.go]: Yup, fair, not sure how I missed that.

**Review by bbrks — APPROVED** [review]
LGTM

**Pull request merged**